### PR TITLE
C396373 Save invoice fiscal year after adding adjustment on invoice level if FY was undefined and pay against previous FY (thunderjet) (TaaS)

### PR DESCRIPTION
**Pull request by kirstaVova**
***
[C396373](https://foliotest.testrail.io/index.php?/cases/view/396373)
***
[Allure Report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/5038/allure/)
***
Screenshot_341

### CODE REVIEW

File: cypress/e2e/invoices/save-invoise-fiscal-year-after-adding-adjustment.cy.js [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    ]).then((userProperties) => {
+      user = userProperties;
+      cy.login(userProperties.username, userProperties.password, {
+        path: TopMenu.invoicesPath,
+        waiter: Invoices.waitLoading,
+      });
+    });
+  });
+
+  //   after(() => {
```
Comment (dnovikau) [inline cypress/e2e/invoices/save-invoise-fiscal-year-after-adding-adjustment.cy.js:185]: Please uncomment 'after' hook

File: cypress/e2e/invoices/save-invoise-fiscal-year-after-adding-adjustment.cy.js [hunk 2]
```diff
@@ -0,0 +1,295 @@
+import permissions from '../../support/dictionary/permissions';
```
Comment (dnovikau) [inline cypress/e2e/invoices/save-invoise-fiscal-year-after-adding-adjustment.cy.js:1]: Please use one line import for {Permission, TestTypes and DevTeams}

File: cypress/e2e/invoices/save-invoise-fiscal-year-after-adding-adjustment.cy.js [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+      Ledgers.createViaApi(defaultLedger).then((ledgerResponse) => {
+        defaultLedger.id = ledgerResponse.id;
+        defaultFund.ledgerId = defaultLedger.id;
+
+        Funds.createViaApi(defaultFund).then((fundResponse) => {
+          defaultFund.id = fundResponse.fund.id;
+
+          cy.loginAsAdmin({ path: TopMenu.fundPath, waiter: Funds.waitLoading });
+          FinanceHelp.searchByName(defaultFund.name);
+          Funds.selectFund(defaultFund.name);
```
Comment (dnovikau) [inline cypress/e2e/invoices/save-invoise-fiscal-year-after-adding-adjustment.cy.js:85]: Please replace UI actions with API calls in preconditions. To see an example please look at 'fiscal-year-is-not-displayed.cy'

File: cypress/support/fragments/finance/fiscalYears/fiscalYears.js [hunk 4]
```diff
@@ -135,6 +135,7 @@ export default {
       TextField({ name: 'periodEnd' }).fillIn(periodEnd),
       saveAndClose.click(),
     ]);
+    cy.wait(6000);
```
[the remark this answers is not in the document]
Comment (kirstaVova) [reply cypress/support/fragments/finance/fiscalYears/fiscalYears.js:138]: removed

**Review by IhorBohdan — APPROVED** [review]

**Review by TetianaParanich — APPROVED** [review]

**Review by Sherzod-Kenjaev — APPROVED** [review]

**Pull request merged**